src/appendices/working-within-test-code.rst

```diff
@@ -0,0 +1,200 @@
+
+To run your tests, type ``npm test`` in the command line of the terminal.
+Since we are using replit, that means type this below any output in the very right screen of the replit IDE.
+
+.. sourcecode:: bash
+   
+   node index
+   Hello, World!
+   npm test
+
+If your output screen looks like this, hit enter and your tests will run.
+
+If you have written the correct code, the tests will pass.  
+When a test runs, the console will log information about the test.  The dot in **Line 6** is a quick indicator that we passed. 
+In replit, this will be green for a pass.
+
+
+.. sourcecode:: bash 
+   :linenos:
+   
+   > working-within-tests@1.0.0 test /home/runner/Working-Within-Tests-Demo
+   > jasmine
+
+   Randomized with seed 21669
+   Started
+   .
+
+
+   1 spec, 0 failures
+   Finished in 0.008 seconds
+   Randomized with seed 21669 (jasmine --random=true --seed=21669)
+
+
+
+What happens if you fail a test?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Return to the ``hello.js`` file.  Replace the string inside the ``hello`` function with something that is *not* "Hello, World!" 
+Maybe say hello to your friend or pet? 
+
+.. admonition:: note
+
+   "Hello, Coder!" was used for demoing the failed test.
+
+   If your console is too crowded, you can clear it in two ways.  Either clear it by clicking the "X" in the top right corner or type ``clear`` directly into the console.
+
+Once you have changed the string, run the program.
+
+.. sourcecode:: bash
+
+   node index
+   Hello, Coder!
+
+The program ran beautifully.  As it should.  We updated the input that will be printed to the console.
+Let's check our output with the tests.  Type ``npm test`` directly into the console.  What did you get?  
+In replit, the ``F`` on **line 6** will be red to better stand out.  
+
+.. sourcecode:: bash
+   :linenos:
+
+   > working-within-tests@1.0.0 test /home/runner/Working-Within-Tests-Demo
+   > jasmine
+
+   Randomized with seed 04579
+   Started
+   F
+
+   Failures:
+   1) Test Example Solution outputs the correct message
+   Message:
+      Expected 'Hello, Coder!' to be 'Hello, World!'.
+   Stack:
+      Error: Expected 'Hello, Coder!' to be 'Hello, World!'.
+         at <Jasmine>
+         at UserContext.<anonymous> (/home/runner/Working-Within-Tests-Demo/spec/hello.spec.js:10:19)
+         at <Jasmine>
+
+   1 spec, 1 failure
+   Finished in 0.01 seconds
+   Randomized with seed 04579 (jasmine --random=true --seed=04579)
+   npm ERR! Test failed.  See above for more details.
+
+This is great!  Output like this can help you troubleshoot any potential code breakage or deviation.  
+The message tells us exactly what needes to be changed.  Change the string back to ``Hello, World!`` and run the tests again.
+
+Congrats! You passed! 
+
+Why Tests?
+----------
+
+The code still worked when we changed the string from ``Hello, World!`` to ``Hello, Coder!`` so why test it? 
+That is a very good question that we will explore in a later chapter. 
```

Comment:
What about "For now, you just need to know that tests are written to see if the code meets the requirements of the instructions. We will be covering more on how we know what to test in a later chapter."?

Reply:
yes, this is better. ;D using this.  thanks! :)